### PR TITLE
treat / as root for restore of onedrive and sharepoint

### PR DESCRIPTION
**Pull request by neha-Gupta1**
passing '/' will select anything for backup details and restore for onedrive and sharepoint

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #https://github.com/alcionai/corso/issues/3252
#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/cli/utils/flags.go [hunk 1]
```diff
@@ -215,6 +215,11 @@ func trimFolderSlash(folders []string) []string {
 	res := make([]string, 0, len(folders))
 
 	for _, p := range folders {
+		if len(p) == 1 && p[0] == path.PathSeparator {
```
Comment (ryanfkeepers) [inline src/cli/utils/flags.go:218]: This is an overcomplicated form of equals:
```suggestion
		if p == path.PathSeparator {
```

File: src/cli/utils/onedrive_test.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ func (suite *OneDriveUtilsSuite) TestIncludeOneDriveRestoreDataSelectors() {
 		containsOnly      = []string{"contains"}
 		prefixOnly        = []string{"/prefix"}
 		containsAndPrefix = []string{"contains", "/prefix"}
+		onlySlash         = []string{"/"}
```
Comment (ryanfkeepers) [inline src/cli/utils/onedrive_test.go:29]: If we compare on a const, we should also test on a const.
```suggestion
		onlySlash         = []string{path.PathSeparator}
```

File: src/cli/utils/sharepoint_test.go [hunk 3]
```diff
@@ -182,6 +183,13 @@ func (suite *SharePointUtilsSuite) TestIncludeSharePointRestoreDataSelectors() {
 			},
 			expectIncludeLen: 2,
 		},
+		{
+			name: "only / inputs",
```
Comment (ryanfkeepers) [inline src/cli/utils/sharepoint_test.go:187]:
```suggestion
			name: "folder with just /",
```

**Review by ashmrtn — COMMENTED** [review]
Changes LGTM but will let Keepers approve. Could you also add test cases to the CLI-level selector stuff for both OneDrive and SharePoint please? The cases will look a lot like the `AllFiles` [ones](https://github.com/alcionai/corso/blob/1dcdd28a90b2fbe1185b68a5ea1745fc30657b9a/src/cli/utils/testdata/opts.go#L353) but the input will be just `/` instead of `selectors.Any()`

**Review by ryanfkeepers — APPROVED** [review]

**Pull request merged**